pkg/iris/api.go

```diff
@@ -46,16 +57,18 @@ func (a *api) GetConfiguration(configurationName string, accessToken string) ([]
 }
 
 func (a *api) PostResults(results []byte, accessToken string) error {
-	resultUrl := fmt.Sprintf("https://%s/services/vsm-iris/v1/results", a.uri)
+	resultUrl := fmt.Sprintf("%s/services/vsm-iris/v1/results", a.uri)
 	postReq, err := http.NewRequest("POST", resultUrl, nil)
-	postReq.Header.Set("Content-Type", "application/json")
-	postReq.Header.Set("Authorization", "Bearer "+accessToken)
-	postReq.Body = ioutil.NopCloser(bytes.NewBuffer(results))
 	if err != nil {
-		log.Errorf("Error while posting results: %v", err)
+		log.Errorf("Error creating request to post results results: %v", err)
 		return err
 	}
-	resp, err := http.DefaultClient.Do(postReq)
+	postReq.Header.Set("Content-Type", "application/json")
```

Comment:
do you think it is a good idea to put this generic logic into a generic function just to reuse it across different apis.

Reply:
Set("Content-Type", "application/json")